src/Razor/src/Microsoft.AspNetCore.Razor.Language/RazorProject.cs

```diff
@@ -56,6 +70,8 @@ public IEnumerable<RazorProjectItem> FindHierarchicalItems(string path, string f
         /// traverses to the <paramref name="basePath"/>.
         /// e.g.
         /// (/Views, /Views/Home/View.cshtml) -> [ /Views/Home/FileName.cshtml, /Views/FileName.cshtml ]
+        /// 
+        /// Project items returned by this method have inferred FileKinds from their corresponding file paths.
```

Comment:
Do we use this API? I'm a little concerned about APIs like this that can't be used correctly.

Reply:
We use it for hierarchical imports (`_ViewImports.cshtml` and `_Imports.razor`). I made the assumption that we wouldn't care about people  marking _ViewImports.cshtml's as components or vice versa with `_Imports.razor`. 

Basically there are some error cases where users could mark files such as `_ViewImports.cshtml` as a component and then indirectly get the `_ViewImports.cshtml` file inclusions in other non-marked .cshtml files. Vice versa being true with `_Imports.razor`. Those incorrect behaviors aside I figured the simplicity of the fix in this PR warranted the slight drawback of ignoring those two uncommon error cases.